### PR TITLE
Start `ui-tests`

### PR DESCRIPTION
**Pull request by jtpio**
Fixes https://github.com/jupyterlite/xeus-python-kernel/issues/29

- [x] Add `ui-tests` top-level folder
- [x] Test on `chromium` and `firefox`
- [x] Snapshot test to check the kernel is correctly registered
- [x] Test execute code
- [x] Test access files
- [x] ~Try `webkit` to investigate https://github.com/jupyterlite/xeus-python-kernel/issues/152~ -> will try in a separate PR
- [x] Add workflow to update snapshots

### CODE REVIEW

**Comment by jtpio** [comment]
Deferring the `webkit` investigation to a separate PR.

File: .eslintignore [hunk 1]
```diff
@@ -5,3 +5,6 @@ coverage
 tests
 src/worker.ts
 src/web_worker_kernel.ts
+
+# TODO: remove
```
Comment (jtpio) [inline .eslintignore:9]: Ah I missed this one.

Happy to track it in a separate issue to not block this PR if that's fine.

File: ui-tests/test/general.test.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+test.use({ waitForApplication: firefoxWaitForApplication });
+
+test.describe('General', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto('lab/index.html');
+  });
+
+  test('Kernel is registered', async ({ page }) => {
+    const imageName = 'launcher.png';
+    expect(await page.screenshot()).toMatchSnapshot(imageName);
```
Comment (martinRenou) [inline ui-tests/test/general.test.ts:19]: Should we reduce the scope of the screenshot and just take the XPython item in the UI? That way the test is less likely to fail in the futur.

**Review by martinRenou — COMMENTED** [review]
Amazing! Thanks

File: ui-tests/test/general.test.ts (hunk 2 above)
Comment (jtpio) [reply ui-tests/test/general.test.ts:19]: Maybe just the launcher then?

Comment (martinRenou) [reply ui-tests/test/general.test.ts:19]: We could even just make sure the test XPython is in the UI and. Not take a screenshot.

Comment (jtpio) [reply ui-tests/test/general.test.ts]: Right. Although with a screenshot we can also check it appears under the Console section, and its display name is correct (but these could also be regular expects).

**Review by martinRenou — APPROVED** [review]
Thanks!

**Pull request merged**